packages/compass-aggregations/src/components/pipeline-toolbar/pipeline-header/pipeline-stages.tsx

```diff
@@ -97,11 +98,12 @@ export const PipelineStages: React.FunctionComponent<PipelineStagesProps> = ({
 };
 
 const mapState = (state: RootState) => {
-  const stages = getPipelineStageOperatorsFromBuilderState(state);
+  const stages = getPipelineStageOperatorsFromBuilderState(state, false);
   const isResultsMode = state.workspace === 'results';
+  const isStageMode = state.pipelineBuilder.pipelineMode === 'builder-ui';
   return {
-    stages,
-    showAddNewStage: !isResultsMode && stages.length === 0,
+    stages: stages.filter(Boolean) as string[],
```

Comment:
With the current state of this PR, passing `true` to `getPipelineStageOperatorsFromBuilderState()` in the lines above would already apply the same filter and give the right type automatically, right? What am I missing? :)

Reply:
for computing `showAddNewStage` prop, we take into consideration the null values as well - which means there's a stage added already (which does not have an operator yet) and in that case we hide `Add New Stage` button.